### PR TITLE
Add connected papers integration

### PR DESCRIPTION
**Pull request by KanHarI**
Added connected papers integration based on PwC integration, + functional "off" switch

### CODE REVIEW

**Comment by KanHarI** [comment]
Sorry for previously creating PRs based on wrong branch

**Comment by KanHarI** [comment]
For a functional demo, please check out https://arxiv.beta.connectedpapers.com:5001/

File: browse/static/js/connectedpapers.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  $output.html('<p>Loading...</p>');
+
+  const REST_ADDR = 'https://rest.migration.connectedpapers.com/';
+  const CONNECTED_PAPERS_ADDR = 'https://www.connectedpapers.com/';
+  const ARXIV_THUMBNAILS_ADDR = CONNECTED_PAPERS_ADDR + 'arxiv_thumbnails/';
+  const NUMBER_OF_THUMBNAILS = 18;
+  
+  var arxivId = window.location.pathname.split('/').reverse()[0];
+  var arxivIdToCPIdUrl = REST_ADDR + '?arxiv=' + arxivId;
+  var communicationErrorHtml = '<p>Oops, seems like communiation with the Connected Papers server is down.</p>';
```
Comment (mhl10) [inline browse/static/js/connectedpapers.js:19]:
```suggestion
  var communicationErrorHtml = '<p>Oops, seems like communication with the Connected Papers server is down.</p>';
```

**Comment by mhl10** [comment]
Hi @KanHarI, your changes look good. Thank you for creating the functional demo; I'll review this with the team tomorrow.

**Comment by KanHarI** [comment]
> Hi @KanHarI, your changes look good. Thank you for creating the functional demo; I'll review this with the team tomorrow.

Thanks!

File: browse/static/js/connectedpapers.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        // Toggled off
+        return;
+      }
+
+      var graphUrl = CONNECTED_PAPERS_ADDR + 'main/' + paperId + '/arxiv';
+      var buildGraphLinkHtml = '<a href="' + graphUrl + '" target="_blank"><p style="margin:0;">View graph for ' + 
+                                title + '</p></a>';
+      var seeGraphLinkHtml = '<a href="' + graphUrl + '" target="_blank"><p style="margin:0;">View graph for ' +
+                              title + '</p></a>';
+      var graphNotVisual = '<p>Seems like ' + title + ' is still not in our database. Please try again in a few days.</p>';
```
Comment (SBBCornell) [inline browse/static/js/connectedpapers.js:53]: Suggest not using the title in this 'not found' message, as arXiv paper titles can be very long! The message might get lost. "Seems like this paper is still not in our database" seems sufficient to me.

File: browse/static/js/connectedpapers.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    var versionsFetchUrl = REST_ADDR + '?versions=' + paperId;
+
+    $.get(versionsFetchUrl).done(versionsResponse => {
+      if ($output.html() == '') {
+        // Toggled off
+        return;
+      }
+
+      var graphUrl = CONNECTED_PAPERS_ADDR + 'main/' + paperId + '/arxiv';
+      var buildGraphLinkHtml = '<a href="' + graphUrl + '" target="_blank"><p style="margin:0;">View graph for ' +
```
Comment (SBBCornell) [inline browse/static/js/connectedpapers.js:49]: I'd like to suggest a little additional "found" language here. Something like "Connection found. View graph for..."

We are trying to think ahead to when an arxiv user might have multiple services toggles on in each tab. We think leading with a clear "found" message will help users quickly identify relevant info.

File: browse/static/js/connectedpapers.js (hunk 2 above)
Comment (SBBCornell) [reply browse/static/js/connectedpapers.js:53]: Also, if there is an option for users to submit papers this would be a good place to add that link.

Comment (eddiesmo) [reply browse/static/js/connectedpapers.js:53]: Thanks for the suggestion Shamsi - I agree. Itay, can you please update the microcopy?

Re submitting papers - we don't have this option currently.

File: browse/static/js/connectedpapers.js (hunk 3 above)
Comment (eddiesmo) [reply browse/static/js/connectedpapers.js:49]: Thanks for the comment Shamsi!

When a user clicks a paper sometimes the graph will be built for the first time, just for them (we can't build graphs for all existing papers in advance). In rare cases there could be a problem and there will be no graph. Therefore we wouldn't want to promise users that papers were "found" before we actually found them. 

That said we agree that the microcopy here could be more inviting and we suggest the following change:
image
(in contrast to what we had now:)

image

How does that look to you?
Thanks!

File: browse/static/js/connectedpapers.js (hunk 2 above)
Comment (KanHarI) [reply browse/static/js/connectedpapers.js]: Done.

File: browse/static/js/connectedpapers.js (hunk 3 above)
Comment (KanHarI) [reply browse/static/js/connectedpapers.js]: Hi, I've updated the PR with the design Eddie posted above. Please let us know if you have additional feedback.

**Comment by KanHarI** [comment]
Hi! we have updated our functional demo at 
https://arxiv.beta.connectedpapers.com:5001/abs/0704.0526

 Feel free to test it!

**Comment by KanHarI** [comment]
Hi, updated our demo again - please test it :)
https://arxiv.beta.connectedpapers.com:5001/abs/0704.0526

**Review by mhl10 — APPROVED** [review]

**Pull request merged**